js/core/option_manager.js

```diff
@@ -0,0 +1,215 @@
+import coreDataUtils from "./utils/data";
+import typeUtils from "./utils/type";
+import compareUtils from "./utils/compare";
+import CallBacks from "./utils/callbacks";
+import { extend } from "./utils/extend";
+
+export class OptionManager {
+    constructor(options, optionsByReference, deprecatedOptions) {
+        this._options = options;
+        this._optionsByReference = optionsByReference;
+        this._deprecatedOptions = deprecatedOptions;
+        this._changingCallbacks = CallBacks({ syncStrategy: true });
+        this._changedCallbacks = CallBacks({ syncStrategy: true });
+        this._deprecatedCallbacks = CallBacks({ syncStrategy: true });
+        this._cachedDeprecateNames = [];
+        this.cachedGetters = {};
+        this.cachedSetters = {};
+    }
+
+    _logWarningIfDeprecated(option) {
```

Comment:
Why you renamed the callback but left the `log...` name for the method? I suggest something like `notifyDeprecated`.